ml_model/src/main/java/edu/wm/sealab/featureextraction/FeatureVisitor.java

```diff
@@ -234,6 +256,19 @@ public void visit(DoubleLiteralExpr dle, Void arg) {
     features.incrementNumOfLiterals();
   }
 
+  /**
+   * This method identifies comments in a java method and sums them up to the total number of
+   * comments
+   *
+   * @param ClassOrInterfaceDeclaration
+   * @param Void
+   */
+  @Override
+  public void visit(ClassOrInterfaceDeclaration cu, Void arg) {
+    super.visit(cu, arg);
+    features.setNumOfComments(cu.getAllContainedComments().size());
```

Comment:
This counts the number of comment AST nodes. I'm not sure that's the feature we want: I suspect it's more likely that we're looking for "number of lines with a comment". @NGimhana can you comment on this? What is the actual definition of the comments feature?

Reply:
#comments feature should include all the comments that belong to the method (say JavaDoc, line, block and all the orphan comments as well). If we just consider only the "number of lines with a comment" then it discards orphan comments like below. So thats why here we count the number of comment AST nodes. Refer to point 4 here- https://github.com/sea-lab-wm/complexity-verification-project/pull/10#pullrequestreview-1553418229 

```
if (provider.isSupported(context)) {
        // provider is compatible! proceeding
       
        System.out.println(...)
```